openedx_external_enrollments/tests/external_enrollments/tests_mit_hz_external_enrollments.py

```diff
@@ -77,3 +81,67 @@ def test_get_user_id(self):
             self.base._get_user_id(email),  # pylint: disable=protected-access
             expected_data,
         )
+
+    @patch(
+        'openedx_external_enrollments.external_enrollments.'
+        'mit_hz_external_enrollment.MITHzInstanceExternalEnrollment._check_user_subscription'
+    )
+    def test_get_course_home_url(self, get_course_home_url_mock):
+        """
+        Test _get_course_home_url method.
+        """
+        get_course_home_url_mock.return_value = None
+
+        self.base._get_course_home_url({}, {})  # pylint: disable=protected-access
+
+        get_course_home_url_mock.assert_called_once()
+
+    @patch('openedx_external_enrollments.external_enrollments.mit_hz_external_enrollment.ExternalEnrollment')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.'
+        'mit_hz_external_enrollment.MITHzInstanceExternalEnrollment._check_user'
+    )
+    def test_check_user_subscription_with_existing_enrollment(self, check_user_mock, model_mock):
+        """
+        Test _check_user_subscription method with existing enrollment object.
+        """
+        data = {
+            'course_id': 'course_id',
+            'user_email': 'user_email',
+        }
+        model_mock.objects.get.return_value = Mock()
+
+        self.base._check_user_subscription(data)  # pylint: disable=protected-access
+
+        model_mock.objects.get.assert_called_once_with(
+            controller_name='mit_hz',
+            course_shell=data.get('course_id'),
+            email=data.get('user_email'),
+            meta={},
+        )
+        check_user_mock.assert_called_once()
+
+    @patch('openedx_external_enrollments.external_enrollments.mit_hz_external_enrollment.ExternalEnrollment')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.'
+        'mit_hz_external_enrollment.MITHzInstanceExternalEnrollment._check_user'
+    )
+    def test_check_user_subscription_with_non_existing_enrollment(self, check_user_mock, model_mock):
```

Comment:
Check that "save" is not called. @ivanvgh

Reply:
Since the save method comes from the `ExternalEnrollment.objects.get` instance and in this case, it has a side effect, I didn't find a way to mock that instance because it does not have a value, but I've added the return assertion and we have the
`check_user_mock.assert_not_called()`. Are we good with it?